ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/QueryStreamWriter.java

```diff
@@ -125,6 +125,8 @@ public void write(final OutputStream out) {
 
       if (limitReached) {
         objectMapper.writeValue(out, StreamedRow.finalMessage("Limit Reached"));
+      } else if (complete) {
+        objectMapper.writeValue(out, StreamedRow.finalMessage("Query Completed"));
```

Comment:
Thanks for the fix @AlanConfluent! Can we add test coverage for this?

Reply:
Sure, added some unit test coverage for both code paths.